File: CommonBase/src/main/java/com/common/data/ResultData.java
package com.common.data;

import java.io.Serializable;

/**
 * 服务器响应的结果数据
 *
 * @param <T>
 */
public class ResultData<T> implements Serializable {

    /**
     * 状态码
     */
    public int code;

    /**
     * 描述
     */
    public String desc;

    /**
     * 操作结果
     */
    public boolean ok;

    /**
     * 数据
     */
    public T data;

    /**
     * 状态描述
     */
    public State state;

    public class State {
        public int    code;
        public String desc;
    }

    @Override
    public String toString() {
        return "ResultData{" + "code=" + code + ", desc='" + desc + '\'' + ", ok=" + ok + ", data=" + data + '}';
    }
}

File: CommonBase/src/main/java/com/common/data/ResultData.kt
package com.common.data

import java.io.Serializable

/**
 * 服务器响应的结果数据
 *
 * @param <T>
</T> */
class ResultData<T> : Serializable {
    /**
     * 状态码
     */
    var code = 0

    /**
     * 描述
     */
    var desc: String? = null

    /**
     * 操作结果
     */
    var ok = false

    /**
     * 数据
     */
    @JvmField
    var data: T? = null

    /**
     * 状态描述
     */
    var state: State? = null

    inner class State {
        var code = 0
        var desc: String? = null
    }

    override fun toString(): String {
        return "ResultData{code=$code, desc='$desc', ok=$ok, data=$data}"
    }
}